nodogsplash2/files/nodogsplash.init

```diff
@@ -248,10 +248,42 @@ create_instance() {
   procd_close_instance
 }
 
+depends() {
+  if [ "$1" = "iptables" ] ; then
+    if [ $(uci get nodogsplash.@nodogsplash[0].fwhook_enabled) = "1" ] ; then
+      if $WD_DIR/ndsctl status > /dev/null; then
+       echo " * Restarting NodogSplash" 
+       /etc/init.d/nodogsplash restart
+      fi
+    else
+      echo " * NodogSplash fwhook is disabled"
+    fi
+  fi
+}
+
 start_service() {
   include /lib/functions
-
   mkdir -p /tmp/etc/
+  
+  if [ $(uci get nodogsplash.@nodogsplash[0].fwhook_enabled) = "1" ] ; then
+    if ! uci get firewall.nodogsplash2.path &> /dev/null ; then
+        if [ -f '/tmp/etc/ndshook.include' ] ; then
+          rm /tmp/etc/ndshook.include
+        fi
```

Comment:
The last indentation is four spaces here. But it should be two.
I think that should be all.